HiP-EventStoreLib/EventStreamExtensions.cs

```diff
@@ -0,0 +1,54 @@
+﻿using PaderbornUniversity.SILab.Hip.EventSourcing.Events;
+using System;
+using System.Threading.Tasks;
+
+namespace PaderbornUniversity.SILab.Hip.EventSourcing
+{
+    public static class EventStreamExtensions
+    {
+        /// <summary>
+        /// Creates the current state of an entity from the event stream
+        /// </summary>
+        /// <typeparam name="T">Type of the resulting object</typeparam>
+        /// <param name="stream">Event stream</param>
+        /// <param name="resourceType">Resource type</param>
+        /// <param name="id"></param>
+        /// <returns>The resulting object</returns>
+        public static async Task<T> GetCurrentObjectFromEventStream<T>(this IEventStream stream, ResourceType resourceType, int id) where T : new()
```

Comment:
I think we can restrict `T` to a reference type (`where T : class, new()`), because it wouldn't even work for value types: Let's say `T` is some struct. Then, a `CreatedEvent` would set `obj = Activator.CreateInstance<T>()`, which is still `default(T)`. So none of the `PropertyChangedEvent`s coming after that would be applied to `obj` because `obj == default(T)`.